packages/router/router.ts

```diff
@@ -939,7 +953,13 @@ export function createRouter(init: RouterInit): Router {
     if (!actionMatch.route.action) {
       result = getMethodNotAllowedResult(location);
     } else {
-      result = await callLoaderOrAction("action", request, actionMatch);
+      result = await callLoaderOrAction(
+        "action",
+        request,
+        actionMatch,
+        matches,
+        router.basename
```

Comment:
We now expect the hierarchy of matches and the basename for all calls to `callLoaderOrAction`, that way we can make changes to the `Response` `Location` header as needed.